File: Closest Strings/Closest Strings.cpp
class Solution{
public:	
	
	    int shortestDistance(vector<string> &s, string word1, string word2)
    {
        int first=-1;
        int second=-1,ans=INT_MAX;;
        
        for(int i=0; i<s.size(); i++)
        {
                if(s[i]==word1)
                first=i;
                
                if(s[i]==word2)
                second=i;
                
                if(first!=-1 && second!=-1)
                {
                    ans=min(ans,abs(first-second));
                }
        }
        
        return ans;
    }


};

File: Closest Strings/Closest Strings.py
class Solution:
    def shortestDistance(s, word1, word2):
        # code here
        if word1==word2:
            return 0
        a=[]
        b=[]
        for i in range(len(s)):
            if s[i]==word1:
                a.append(i)
            elif s[i]==word2:
                b.append(i)
        res=len(s)
        for i in a:
            for j in b:
                res=min(res,abs(i-j))
        return res
    

S=["hi","hey","g","h"]
word1="hi"
word2="h"
print (Solution.shortestDistance(S,word1,word2))